raiden-dapp/src/components/PlannedUdcWithdrawalInformation.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div
+    class="planned-udc-withdrawal-information d-flex justify-center align-center px-2"
+    :style="{ color, backgroundColor }"
+  >
+    <template v-if="noWithdrawInProgress">
+      {{ $t('udc.information.no-withdrawal-in-progress') | upperCase }}
+    </template>
+
+    <template v-else>
+      <img
+        class="planned-udc-withdrawal-information__status-icon mr-2"
+        :src="require(`@/assets/planned-udc-withdrawal/status-icon-${status}.svg`)"
+        :width="iconSize"
+        :height="iconSize"
+      />
+
+      <amount-display
+        class="planned-udc-withdrawal-information__amount"
+        :token="udcToken"
+        :amount="plannedWithdrawal.amount"
+      />
+
+      <v-spacer />
+
+      <span v-if="isPending" class="planned-udc-withdrawal-information__pending-message">
+        {{ remainingBlocksUntilReady }}
+        {{ $t('udc.information.blocks-remaining') | upperCase }}
+      </span>
+
+      <a
+        v-if="isReady"
+        class="planned-udc-withdrawal-information__navigation-link mr-2"
+        :style="{ color }"
+        @click="navigateToWithdrawal"
+      >
+        {{ $t('udc.information.confirmed-navigation-link') | upperCase }}
+      </a>
+
+      <img
+        v-if="isReady"
+        class="planned-udc-withdrawal-information__navigation-icon mr-2"
+        :src="require(`@/assets/planned-udc-withdrawal/navigation-arrow.svg`)"
+        :width="iconSize"
+        :height="iconSize"
+        @click="navigateToWithdrawal"
+      />
+    </template>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop } from 'vue-property-decorator';
+import { Token } from '@/model/types';
+import { PlannedUdcWithdrawal } from '@/store/user-deposit-contract';
+import AmountDisplay from '@/components/AmountDisplay.vue';
+import NavigationMixin from '@/mixins/navigation-mixin';
+
+@Component({ components: { AmountDisplay } })
+export default class PlannedUdcWithdrawalInformation extends NavigationMixin {
+  @Prop()
+  plannedWithdrawal!: PlannedUdcWithdrawal | undefined;
+
+  @Prop({ required: true })
+  udcToken!: Token;
+
+  @Prop({ required: true })
+  blockNumber!: number;
+
+  readonly assetSourceDirectory = '@/assets/planned-udc-withdrawal';
+  readonly iconSize = '13.33px';
+
+  get noWithdrawInProgress(): boolean {
+    return this.plannedWithdrawal === undefined;
+  }
+
+  get remainingBlocksUntilReady(): number {
+    return this.plannedWithdrawal !== undefined
+      ? this.plannedWithdrawal.withdrawBlock - this.blockNumber
+      : 0;
+  }
+
+  get isPending(): boolean {
+    return this.plannedWithdrawal !== undefined && this.remainingBlocksUntilReady > 0;
+  }
+
+  get isReady(): boolean {
+    return this.plannedWithdrawal !== undefined && this.remainingBlocksUntilReady <= 0;
+  }
+
+  get color(): string {
```

Comment:
Personally, I prefer the style to stay in the `styles` part of the component and use the conditionals to switch between `class` names instead if possible. Then I know that I only have to bother with the style section if I want to look for style-related stuff. Isn't that the beauty with Vue? :)

Reply:
Hmm. Vue is not opinionated about that at all. But you have a good point. This is a really nice hint and I will definitely implement it. I like it especially, because it allows me to get rid of fore- and background color and group them in the classes. Thank you very much. :hugs:

Reply:
Done. Including an improvement to not split link text and icon (causes to replicate the click function and is stupid). The past issue was simple the vertical alignment. But I solved it better now, resolving in better "logic" and less code.